addons/sale_timesheet/views/project_update_templates.xml

```diff
@@ -31,17 +32,80 @@
+            <tr t-foreach="revenues['data']" t-as="revenue">
+                <td t-out="profitability_labels[revenue['id']]"/>
+                <td class="text-end" t-out="revenue['formatted']['invoiced']"/>
+                <td class="text-end" t-out="revenue['formatted']['to_invoice']"/>
+                <td class="text-end" t-out="revenue['formatted']['expected_revenue']"/>
+            </tr>
+            <tfoot>
+                <td class="fw-bolder text-end">Total</td>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_invoiced']"/>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_to_invoice']"/>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_revenue']"/>
+            </tfoot>
+        </tbody>
+    </table>
+
+    <table class="table table-striped mt-4">
+        <thead class="border-2 border-start-0 border-end-0">
+            <tr>
+                <th class="fw-bolder">Costs</th>
+                <th class="fw-bolder text-end">Billed</th>
+                <th class="fw-bolder text-end">To Bill</th>
+                <th class="fw-bolder text-end">Expected</th>
+            </tr>
+        </thead>
+        <tbody>
+            <t t-set="costs" t-value="profitability['profitability_items']['costs']"/>
+            <tr t-foreach="costs['data']" t-as="cost">
+                <td t-out="profitability_labels[cost['id']]"/>
+                <td class="text-end" t-out="cost['formatted']['billed']"/>
+                <td class="text-end" t-out="cost['formatted']['to_bill']"/>
+                <td class="text-end" t-out="cost['formatted']['expected_cost']"/>
+            </tr>
+            <tfoot>
+                <td class="fw-bolder text-end">Total</td>
+                <td class="fw-bolder text-end" t-out="profitability['costs_formatted']['total_billed']"/>
+                <td class="fw-bolder text-end" t-out="profitability['costs_formatted']['total_to_billed']"/>
+                <td class="fw-bolder text-end" t-out="profitability['costs_formatted']['total_cost']"/>
+            </tfoot>
+        </tbody>
+    </table>
+
+    <table class="table table-sm mt-4">
+        <tr>
+            <td class="fw-bolder">Margin</td>
+            <td t-attf-class="#{'text-danger' if profitability['costs'] &lt; 0 else 'text-success'}" style="text-align: right; font-weight: bolder"><t t-out="profitability['costs_formatted']['total_cost']"/></td>
+            <td t-attf-class="#{'text-danger' if profitability['revenues'] &lt; 0 else 'text-success'}" style="text-align: right; font-weight: bolder"><t t-out="profitability['revenues_formatted']['total_revenue']"/></td>
+            <td t-attf-class="#{'text-danger' if profitability['margin'] &lt; 0 else 'text-success'}" style="text-align: right; font-weight: bolder"><t t-out="profitability['margin_formatted']"/></td>
+        </tr>
+    </table>
+</div>
```

Comment:
Are you sure the indentation is correctly managed? If you see above your changes, there is no indentation, there is probably why we have no indentation. 🤔

Reply:
> Are you sure the indentation is correctly managed? If you see above your changes, there is no indentation, there is probably why we have no indentation. 🤔

 I have done it without indentation. here already added a comment.  [link](https://github.com/odoo/odoo/blob/17.0/addons/project/views/project_update_templates.xml#L16) thanks